narwhals/_pandas_like/utils.py

```diff
@@ -352,7 +354,11 @@ def non_object_native_to_narwhals_dtype(dtype: str, version: Version) -> DType:
         return dtypes.String()
     if dtype in {"bool", "boolean", "boolean[pyarrow]", "bool[pyarrow]"}:
         return dtypes.Boolean()
-    if dtype == "category" or dtype.startswith("dictionary<"):
+    if dtype.startswith("dictionary<"):
+        return dtypes.Categorical()
+    if dtype == "category":
+        if native_dtype.ordered:
+            return dtypes.Enum(categories=native_dtype.categories)
         return dtypes.Categorical()
```

Comment:
this would be a breaking change, so i'm not totally sure about it - could we preserve the current behaviour in `v1` and only make this change in the main namespace? the `version` variable is available in this function, you can use that